src/main.rs

```diff
@@ -285,42 +315,57 @@ fn print_tree<'a>(package: &'a PackageId,
 fn print_dependency<'a>(package: &'a PackageId,
                         kind: Kind,
                         graph: &Graph<'a>,
+                        format: &str,
                         direction: EdgeDirection,
                         symbols: &Symbols,
                         visited_deps: &mut HashSet<&'a PackageId>,
                         levels_continue: &mut Vec<bool>,
                         all: bool) {
-    if let Some((&last_continues, rest)) = levels_continue.split_last() {
-        for &continues in rest {
-            let c = if continues {
-                symbols.down
-            } else {
-                " "
-            };
-            print!("{}   ", c);
-        }
-
-        let c = if last_continues {
-            symbols.tee
-        } else {
-            symbols.ell
-        };
-        print!("{0}{1}{1} ", c, symbols.right);
-    }
-
     let new = all || visited_deps.insert(package);
     let star = if new {
         ""
     } else {
         " (*)"
     };
 
-    println!("{}{}", package, star);
-
     if !new {
         return;
     }
 
+    if symbols.indent {
+        if let Some((&last_continues, rest)) = levels_continue.split_last() {
+            for &continues in rest {
+                let c = if continues {
+                    symbols.down
+                } else {
+                    " "
+                };
+                print!("{}   ", c);
+            }
+
+            let c = if last_continues {
+                symbols.tee
+            } else {
+                symbols.ell
+            };
+            print!("{0}{1}{1} ", c, symbols.right);
+        }
+    }
+
+    let metadata = graph.node_metadata.get(package).unwrap().clone();
```

Comment:
Can you move this message format logic out to a separate function? As we add more substitution patterns it'll get more complicated.